transmat/zip/zip_unpack_test.go

```diff
@@ -0,0 +1,124 @@
+package ziptrans
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	. "github.com/smartystreets/goconvey/convey"
+
+	api "go.polydawn.net/go-timeless-api"
+	"go.polydawn.net/go-timeless-api/rio"
+	"go.polydawn.net/rio/fs"
+	"go.polydawn.net/rio/fs/osfs"
+	"go.polydawn.net/rio/fsOp"
+	"go.polydawn.net/rio/testutil"
+	"go.polydawn.net/rio/transmat/mixins/tests"
+)
+
+func TestZipUnpack(t *testing.T) {
+	Convey("Spec compliance: Zip unpack", t,
+		testutil.Requires(testutil.RequiresCanManageOwnership, func() {
+			Convey("Using kvfs warehouse, in content-addressable mode:", func() {
+				testutil.WithTmpdir(func(tmpDir fs.AbsolutePath) {
+					osfs.New(tmpDir).Mkdir(fs.MustRelPath("bounce"), 0755)
+					tests.CheckRoundTrip(PackType, Pack, Unpack, api.WarehouseLocation(fmt.Sprintf("ca+file://%s/bounce", tmpDir)))
+					// Following tests could be done in all modes, but isn't about warehouses, so would be redundant to do so.
+					tests.CheckCachePopulation(PackType, Pack, Unpack, api.WarehouseLocation(fmt.Sprintf("ca+file://%s/bounce", tmpDir)))
+				})
+			})
+			Convey("Using kvfs warehouse, in *non*-content-addressable mode:", func() {
+				testutil.WithTmpdir(func(tmpDir fs.AbsolutePath) {
+					tests.CheckRoundTrip(PackType, Pack, Unpack, api.WarehouseLocation(fmt.Sprintf("file://%s/bounce", tmpDir)))
+				})
+			})
+		}),
+	)
+}
+
+/*
+	Tests against pre-generated, known fixtures of zip binary blobs.
+
+	These tests allow us to cover compat with other zip impls, compression, etc.
+*/
+func TestZipFixtureUnpack(t *testing.T) {
+	Convey("Zip transmat: unpacking of fixtures", t,
+		testutil.Requires(testutil.RequiresCanManageOwnership, func() {
+			testutil.WithTmpdir(func(tmpDir fs.AbsolutePath) {
+				/*
```

Comment:
What's up with this commented out test? I'm assuming you want to put it back in before merging?

Reply:
This was a placeholder for finding some other zip implementations to create archives that could be checked for unpacking. Are there other packers worth creating archives with?